util/app/discovery/discovery.go

```diff
@@ -33,7 +33,7 @@ func Discover(ctx context.Context, repoPath string, enableGenerateManifests map[
 	apps := make(map[string]string)
 
 	// Check if it is CMP
-	conn, _, err := DetectConfigManagementPlugin(ctx, repoPath, []string{})
+	conn, _, err := DetectConfigManagementPlugin(ctx, repoPath, []string{}, []string{})
```

Comment:
Could we go ahead and fully parameterize this function? I think I'd want the detection logic to behave the same as the generation logic.

Reply:
The detection logic excludes the environment variables, so it seemed logical to also exclude the tar filters. I'll take another try at this though.

Reply:
All the call trees should terminate in a function with an `ArgoCDRepoServer` receiver. You can access the constant via `s.initConstants.CMPTarExcludedGlobs`.

Reply:
@crenshaw-dev You were right! I just needed some more coffee. 😄